spring-kafka/src/main/java/org/springframework/kafka/core/KafkaOperations2.java

```diff
@@ -0,0 +1,281 @@
+/*
+ * Copyright 2015-2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.kafka.core;
+
+import java.time.Duration;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+
+import org.apache.kafka.clients.consumer.ConsumerGroupMetadata;
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.clients.consumer.ConsumerRecords;
+import org.apache.kafka.clients.consumer.OffsetAndMetadata;
+import org.apache.kafka.clients.producer.Producer;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.apache.kafka.common.Metric;
+import org.apache.kafka.common.MetricName;
+import org.apache.kafka.common.PartitionInfo;
+import org.apache.kafka.common.TopicPartition;
+
+import org.springframework.kafka.core.KafkaOperations.OperationsCallback;
+import org.springframework.kafka.core.KafkaOperations.ProducerCallback;
+import org.springframework.kafka.support.SendResult;
+import org.springframework.kafka.support.TopicPartitionOffset;
+import org.springframework.lang.Nullable;
+import org.springframework.messaging.Message;
+
+/**
+ * The basic Kafka operations contract returning {@link CompletableFuture}s.
+ *
+ * @param <K> the key type.
+ * @param <V> the value type.
+ *
+ * @author Gary Russell
+ * @since 2.9
+ * @deprecated no longer needed; use {@code KafkaOperations}.
+ */
+@Deprecated
+public interface KafkaOperations2<K, V> {
```

Comment:
Why cannot we remove it altogether?
I believe there is going to a migration from `2.8.x` to `3.0.0` mostly through Spring Boot, so that replacement in the API of the `KafkaOperations` is going to be a breaking change anyway.
Therefore this one is just an extra burden for us to bear for the next version.

What do I miss?

Thanks

Reply:
I was thinking of people that had taken advantage of the migration path provided by 2.9.x...

	KafkaOperations2<Integer, String> templateTx = new KafkaTemplate<>(pfTx).usingCompletableFuture();

This code would still work with 3.0 (with a deprecation warning only).

But, I suppose with a major release, this breaking change might be acceptable.

WDYT?

Reply:
Yes, that's my idea as well: bite the bullet and remove. 
One more breaking change won't hurt as it is already with those replaced return types.